### ISSUE TITLE
Failure when creating resource `dnacenter_configuration_template`

### ISSUE THREAD

**Issue by mschedrin** [cisco-en-programmability/terraform-provider-dnacenter#163]
**Prerequisites**
* [x] Have you tested the operation in the API directly?
* [x] Do you have the latest Terraform provider version?
* [x] Review the [compatibility matrix](https://github.com/cisco-en-programmability/terraform-provider-dnacenter/tree/main#compatibility-matrix) before opening an issue. 

**Describe the bug**
Following error is returned when creating `dnacenter_configuration_template`:

```
│ Error: Failure at CreateTemplate, unexpected response
│ 
│   with dnacenter_configuration_template.day0,
│   on configuration_template.tf line 7, in resource "dnacenter_configuration_template" "day0":
│    7: resource "dnacenter_configuration_template" "day0" {
│ 
╵
```

**Expected behavior**
Resource is created without errors. 

**Environment (please complete the following information):**
* Cisco DNA Center version and patch: 2.3.5.0-70586
* Terraform version: 1.3.6
* Cisco DNA Center provider version: 1.1.1-beta
* OS Version: Ubuntu 20.04

**Additional context**
Debug log [dnac-phase-01.log](https://github.com/cisco-en-programmability/terraform-provider-dnacenter/files/11375642/dnac-phase-01.log)

**Comment by fmunozmiranda** [comment]
Hi @mschedrin, can you verify that the `Day0-by-terraform-4-dev-vesion` template actually exists?

Also, could you try to create a template from API? and then look for it? If you could send me screenshots of the steps in case it is successful, I would greatly appreciate it.

**Comment by mschedrin** [comment]
@fmunozmiranda, the template actually gets created and is visible in the UI after running 'terraform apply'. I did some investigations, and here is what I found:

I have looked into the debug log and see that provider v1.1.2 verifies that the template is created by sending `GET /dna/intent/api/v1/template-programmer/template` (without the template id). Apparently, this API call does not return uncommitted templates, so the provider cannot find the template in the payload and returns an error.

I compared how this worked in v1.0.19. The provider verifies the created template by sending `GET /dna/intent/api/v1/template-programmer/template/920d71f9-43e5-4504-b5f8-8c15731339d5`. It is the same API endpoint, but it includes the ID of the template that has just been created. I have tested it on DNAC 2.3.5, and it works as before. So, if you roll back to the provider behavior from v1.0.19, the resource will hopefully begin to work as expected.
